pystac/catalog.py

```diff
@@ -506,15 +506,10 @@ def get_items(self, *ids: str, recursive: bool = False) -> Iterator[Item]:
                 self.get_stac_objects(pystac.RelType.ITEM),
             )
         else:
-            try:
-                items = chain(
-                    self.get_items(recursive=False),
-                    *(child.get_items(recursive=True) for child in self.get_children()),
-                )
-            except TypeError:
-                # For inherited classes that do not yet support recursive
-                # See https://github.com/stac-utils/pystac-client/issues/485
-                items = self.get_all_items()
```

Comment:
I couldn't figure out how to hit this and it doesn't crop up in pystac-client tests, so maybe it was unreachable? It's hurting my brain to think about it.